setup/fusion/scripts/Comp/colorbleed/switch_ui.py

```diff
@@ -146,16 +148,20 @@ def _refresh(self):
 
     def _on_switch(self):
 
+        asset = self._assets.currentText()
+        _comp = avalon.fusion.get_current_comp()
         if not self._use_current.isChecked():
             file_name = self._comps.itemData(self._comps.currentIndex())
+            # Get the current comp's App (FusionUI)
+            _fusion = _comp.GetApp()
+            # Open the selected comp
+            _fusion.LoadComp(file_name)
         else:
-            comp = avalon.fusion.get_current_comp()
-            file_name = comp.GetAttrs("COMPS_FileName")
-
-        asset = self._assets.currentText()
+            file_name = _comp.GetAttrs("COMPS_FileName")
 
-        import colorbleed.scripts.fusion_switch_shot as switch_shot
-        switch_shot.switch(asset_name=asset, filepath=file_name, new=True)
+        switched = flib.switch(asset)
```

Comment:
Hmm, this wasn't directly clear to me as to what it returns. Is `switched_comp` more readable?